examples/uniswap_v2_usdc_swap.rs

```diff
@@ -0,0 +1,343 @@
+use alloy_provider::ProviderBuilder;
+use alloy_sol_types::{sol, SolCall, SolValue};
+use anyhow::{anyhow, Result};
+use revm::{
+    db::{AlloyDB, CacheDB, EmptyDB, EmptyDBTyped},
+    primitives::{
+        address, keccak256, AccountInfo, Address, Bytes, ExecutionResult, Output, TransactTo, U256,
+    },
+    Database, Evm,
+};
+use std::ops::Div;
+use std::{convert::Infallible, sync::Arc};
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let client = ProviderBuilder::new()
+        .on_reqwest_http(
+            "https://mainnet.infura.io/v3/c60b0bb42f8a4c6481ecd229eddaca27"
+                .parse()
+                .unwrap(),
+        )
+        .unwrap();
+    let client = Arc::new(client);
+    let mut alloydb = AlloyDB::new(client, None);
+    let mut cache_db = CacheDB::new(EmptyDB::default());
```

Comment:
you can just do `CacheDB::new(AlloyDB::new(client, None))`, then calls to e.g. `db.basic` will automatically first get pulled from the alloy provider, and then stored in the cache db! that avoids you needing to do what you are doing on eg L45, L48 etc